### ISSUE TITLE
None does not appear to be an IPv4 or IPv6 address

### ISSUE THREAD

**Issue by felipe3dfx** [jazzband/django-axes#436]
```python
AXES_FAILURE_LIMIT = 10
AXES_USE_USER_AGENT = True
AXES_COOLOFF_TIME = 24
AXES_LOCKOUT_TEMPLATE = '429.html'
AXES_CACHE = 'axes'

CACHES = {
    'axes': {
        'BACKEND': 'django_redis.cache.RedisCache',
        'LOCATION': f'redis://{REDIS_HOST}:{REDIS_PORT}/4',
        'OPTIONS': {
            'CLIENT_CLASS': 'django_redis.client.DefaultClient',
        }
    }
}
```

```python
ValueError: None does not appear to be an IPv4 or IPv6 address
  File "django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "axes/middleware.py", line 46, in __call__
    self.update_request(request)
  File "axes/helpers.py", line 394, in inner
    return func(*args, **kwargs)
  File "axes/middleware.py", line 60, in update_request
    request.axes_ip_address = get_client_ip_address(request)
  File "axes/helpers.py", line 153, in get_client_ip_address
    return str(ip_address(client_ip_address))
  File "python3.6/ipaddress.py", line 54, in ip_address
    address)
```
acme==0.31.0
aioredis==1.2.0
asgiref==3.1.2
asn1crypto==0.24.0
astroid==2.0.4
async-timeout==3.0.1
atomicwrites==1.3.0
attrs==17.4.0
autobahn==19.3.3
Automat==0.6.0
backcall==0.1.0
certbot==0.31.0
certifi==2018.1.18
channels==2.2.0
channels-redis==2.4.0
chardet==3.0.4
Click==7.0
colorama==0.3.7
command-not-found==0.3
ConfigArgParse==0.11.0
configobj==5.0.6
constantly==15.1.0
coverage==4.5.3
cryptography==2.1.4
daphne==2.3.0
decorator==4.4.0
distro-info===0.18ubuntu0.18.04.1
dj-inmemorystorage==1.4.1
Django==2.2.1
django-admin-sortable==2.1.14
django-appconf==1.0.3
django-axes==5.0.2
django-compressor==2.2
django-debug-toolbar==1.11
django-dynamic-fixture==2.0.0
django-extensions==2.1.6
django-ipware==2.1.0
django-redis==4.10.0
django-test-plus==1.2.0
django-widget-tweaks==1.4.3
dodgy==0.1.9
fail2ban==0.10.2
future==0.15.2
gunicorn==19.9.0
h11==0.8.1
hiredis==1.0.0
httplib2==0.9.2
httptools==0.0.13
huey==2.0.1
hyperlink==17.3.1
idna==2.6
incremental==16.10.1
iotop==0.6
ipython==7.5.0
ipython-genutils==0.2.0
isort==4.3.17
jedi==0.13.3
josepy==1.1.0
language-selector==0.1
lazy-object-proxy==1.3.1
mccabe==0.6.1
mock==2.0.0
more-itertools==7.0.0
msgpack==0.6.1
ndg-httpsclient==0.4.4
netifaces==0.10.4
PAM==0.4.2
parsedatetime==2.4
parso==0.4.0
pbr==3.1.1
pep8-naming==0.4.1
pexpect==4.7.0
pickleshare==0.7.5
Pillow==6.0.0
piprot==0.9.10
pluggy==0.9.0
prompt-toolkit==2.0.9
prospector==1.1.6.2
psycopg2-binary==2.8.2
ptyprocess==0.6.0
py==1.8.0
pyasn1==0.4.2
pyasn1-modules==0.2.1
pycodestyle==2.4.0
pydocstyle==3.0.0
pyflakes==1.6.0
Pygments==2.3.1
pygobject==3.26.1
PyHamcrest==1.9.0
PyICU==1.9.8
pyinotify==0.9.6
pylint==2.1.1
pylint-celery==0.3
pylint-django==2.0.2
pylint-flask==0.5
pylint-plugin-utils==0.5
pyOpenSSL==17.5.0
pyRFC3339==1.0
pyserial==3.4
pytest==4.4.1
pytest-cov==2.7.1
pytest-django==3.4.8
pytest-mock==1.10.4
python-apt==1.6.3+ubuntu1
python-debian==0.1.32
python-magic==0.4.15
pytz==2019.1
PyYAML==3.12
raven==6.10.0
rcssmin==1.0.6
redis==3.2.1
requests==2.21.0
requests-futures==0.9.9
requests-toolbelt==0.8.0
requests-unixsocket==0.1.5
requirements-detector==0.6
rjsmin==1.0.12
service-identity==18.1.0
setoptconf==0.2.0
six==1.11.0
snowballstemmer==1.2.1
sqlparse==0.3.0
systemd-python==234
tblib==1.4.0
traitlets==4.3.2
Twisted==19.2.0
txaio==18.8.1
typed-ast==1.3.4
ufw==0.36
unattended-upgrades==0.1
urllib3==1.22
uvicorn==0.7.1
uvloop==0.12.2
wcwidth==0.1.7
websockets==7.0
wrapt==1.11.1
zope.component==4.3.0
zope.event==4.2.0
zope.hookable==4.0.4
zope.interface==4.6.0

**Comment by aleksihakli** [comment]
Hi! What other middleware do you have in your stack and how is the request made? It seems that the `request` object does not have an appropriate IP that could be resolved, which results in the error in Python standard library itself.

You will also have to set `AXES_HANDLER` in order to use a cache:

```python
AXES_HANDLER = 'axes.handlers.cache.AxesCacheHandler'
```

**Comment by aleksihakli** [comment]
@felipe3dfx This behaviour has been fixed in 83a1880c3b88baff8eede713a4114d0429ab78a0 and released in 5.0.3 which is now available on PyPI.

**Comment by aleksihakli** [comment]
Please note that Axes needs to be able to resolve the client address in order for it to function correctly. The documentation has instructions on how to configure Axes and `django-ipware` to look for the correct `request.META` property that contains the user IP :)
